### ISSUE TITLE
About out of discriminator 

### ISSUE THREAD

**Issue by Chenguang-Wang** [SharifAmit/Semi-supervised-Phishing-Detection-GAN#1]
Hello, @SharifAmit
What‘s the meaning of out1(real/adversarial) of Discriminator? I attempt to use this and that in the label to calculate merics, it's bad.
I use out2 to calculate merics but got a higher AUC.

Like this 
print(accuracy, sensitivity, precision, f1, auc)
![捕获](https://user-images.githubusercontent.com/31790598/155154535-318f9ce1-11db-4625-a215-1eb2cb32c10f.PNG)
Thank you!

**Comment by SharifAmit** [comment]
Hi,

**out1** is for calculating mse between real/fake data. We are not classifying between real and fake here, rather measuring error (MSE) between prediction and ground-truth (fake=0 , real=1). This is actually for training the adversarial loss function, specifically Least-Square-GAN loss, given in Equation 1 in the paper. 

The classification of benign and malicious URL is done using **out2**, which uses categorical crossentropy loss, given in equation 2 in the paper.

Thanks

**Comment by Chenguang-Wang** [comment]
Hello，
I can get MSE and the generated urls.
But my computer will crash when I use skimage to calculate SSIM. Is it due to my memory-limit?  
 And how is NRMSE calculated?

Thank you！

**Comment by SharifAmit** [comment]
```
from skimage.metrics import normalized_root_mse
from skimage.metrics import structural_similarity as ssim

nrmse = normalized_root_mse(X_t,pred_gen,normalization='euclidean')
print(nrmse)

ssim = ssim(X_t, pred_gen, data_range=X_t.max()-X_t.min(),multichannel=True)
print(ssim)
```

**Comment by Chenguang-Wang** [comment]
It worked.
Thanks.

**Comment by Chenguang-Wang** [comment]
Hello，
Why is the length of url selected as 200?
Do we have experiments like 128.etc based on the same data?

**Comment by SharifAmit** [comment]
200 is the max url character length we used in the experiment. So the dataset have URLs which are below 200 as well, which were padded in the data preprocessing.  Also, any reliable website targeted for spear phishing won't have very long URLs (500 or 1000 character), which might increase suspicion in people who are being targeted.

Hope this answers your question.

**Comment by Chenguang-Wang** [comment]
<100 100~200 200~300 300~400 >400
good:  25875 104 14 6 1
bad:  22266 5886 1540 183 125

Yes. Bad urls perfer longer.
I mean if 200 is too long on these data. The statistic of good(the first 26000) and bad(the first 30000) is above.
I attempt to use 128 yesterday and get a result very close to 200.

**Comment by SharifAmit** [comment]
Hi,

Thanks for the insights.

If you have any other questions or concerns please reopen the issue.

Thanks

**Comment by westbrrokk** [comment]
> Hello, @SharifAmit What‘s the meaning of out1(real/adversarial) of Discriminator? I attempt to use this and that in the label to calculate merics, it's bad. I use out2 to calculate merics but got a higher AUC.
> 
> Like this print(accuracy, sensitivity, precision, f1, auc) ![捕获](https://user-images.githubusercontent.com/31790598/155154535-318f9ce1-11db-4625-a215-1eb2cb32c10f.PNG) Thank you!



> ###
> **[Chenguang-Wang](/Chenguang-Wang) ** commented [Feb 22, 2022](#issue-1146995899) •

大佬可以问一些问题吗，我是一个小白，
